stl/inc/type_traits

```diff
@@ -2516,6 +2517,41 @@ _NODISCARD constexpr bool _Is_finite(const _Ty _Xx) noexcept { // constexpr isfi
 template <bool _IsConst, class _Ty>
 using _Maybe_const = conditional_t<_IsConst, const _Ty, _Ty>;
 
+template <class _Ty, size_t _Size>
+void _Swap_trivial_arrays(_Ty (&_Left)[_Size], _Ty (&_Right)[_Size]) noexcept {
+    // Swap arrays using temporary buffers of 64 bytes, then doing tail
+    // In optimized build, stack usage would optimize away, and vector registers will be used
+    // 64 is enough to engage AVX-512, still not too much to use SSE2 (eight registers would be used then),
+    // and not too much stack consumption for debug builds where this will not be optimized
+    auto _Left_ptr  = reinterpret_cast<unsigned char*>(&_Left);
+    auto _Right_ptr = reinterpret_cast<unsigned char*>(&_Right);
+
+    constexpr size_t _Part_size_bytes = 64;

```

Comment:
I realize we can detect `/arch` level and provide different values for different levels, but I don't think it worth doing.